data_profiler/data_readers/csv_data.py

```diff
@@ -196,38 +196,39 @@ def is_match(cls, file_path, options=None):
         base_regex = "(?=(?:[^\"]*\"[^\"]*\")*[^\"]*$)"        
         delimiter_regex = re.compile(re.escape(str(delimiter)) + base_regex)
         
-        # Count the possible delimiters 
+        # Count the possible delimiters
         with open(file_path, encoding=file_encoding) as f:
             for line in f:
                 line_count += 1
+                count = 0
                 
                 # Find the location(s) where each delimiter was detected
                 delimiter_locs = [i.start() for i in re.finditer(delimiter_regex, line)]
                 if delimiter:
                     # Count all locations that aren't the last entry
                     # This can cause errors, if you could have a sparse end column
-                    count = 0
                     for loc in delimiter_locs:
-                        # Reason: len(line) - 2; -1 to start count 0 & -1 for last loc
-                        if loc != len(line)-2:
-                            count += 1
+                        count += 1
```

Comment:
for now, why don't just do len(delimiter_locs) ?

Reply:
Fair, PR'd update